llvm/lib/Target/Mips/MipsISelLowering.cpp

```diff
@@ -1046,8 +1046,26 @@ static SDValue attemptOrToIns(SDValue &And, SDValue &Right, SDNode *N,
   if (!(CN = dyn_cast<ConstantSDNode>(And.getOperand(1))))
     return SDValue();
   unsigned Mask0 = ~CN->getZExtValue();
-  if (!isShiftedMask(Mask0, SMPos0, SMSize0))
-    return SDValue();
+
+  KnownBits KnownMaskBits = DAG.computeKnownBits(And.getOperand(1));
```

Comment:
Method ```DAG.computeKnownBits()``` requires an argument of type ```SDValue```, that's the reason why ```And.getOperand(1)``` is passed instead of ```CN```. Maybe we can somehow extract ```SDValue``` from ```CN```?

Reply:
Oh, I didn't spot that because the conversion the other way from SDValue to SDNode is implicit. Yes, you should be able to get the SDValue corresponding to a SDNode's *n*'th result just by calling SDValue(node, *n*).